sap/platform/abap/__init__.py

```diff
@@ -210,8 +219,10 @@ def internal_table_to_xml(abap_table, dest, prefix, row_name_getter=None):
                 dest.write(f'{prefix}<{element}>\n')
                 XMLSerializers.struct_members_to_xml(item, dest, prefix + ' ')
                 dest.write(f'{prefix}</{element}>\n')
-            else:
+            elif isinstance(item, InternalTable):
                 dest.write(f'{prefix}<item>{item}</item>\n')
```

Comment:
The commit message says you use StringTable with <item> but this elif statement checks InternalTable. Could you please explain?

Reply:
My idea is to use StringTable to differentiate between InternalTables with/without the `<item>` tag at the innermost level (where the innermost level table is the one with simple objects (strings) only --> no inner structures/internal tables).

The line you pointed out concerns the table with an inner table (either StringTable or InternalTable), so the table is not at the innermost level. Right now, I realize that the serialization of this table has incorrect implementation. Purely printing `<item>{item}</item>` (where `item` is an InternTable object) results in `<item>[[[ 'item1', 'item2' ]]]</item>`.

Because of that, I suggest wrapping the inner InternalTables to the `<item>` tag in addition to StringTable items. That means that the following XML structure is possible:
```
<ROOT>
 <item>
  <CHILD_TABLE>
   foo
   bar
  </CHILD_TABLE>
 </item>
 <item>
  <CHILD_TABLE>
   baz
   qux
  </CHILD_TABLE>
 </item>
</ROOT>
```
Where `<CHILD_TABLE>` is the inner table of `<ROOT>`. Note, if the `<CHILD_TABLE>` would be an instance of StringTable, its items would be wrapped in the `<item>` tag as well (e.g., `<item>foo</item>`). However, this suggestion is merely my intuition of how it could work. I could not cross-check it with the abapGit, because I could not find an abapGit-generated XML file with such a case (not even sure if it is possible, but for sure is not used anywhere by `sapcli`).

Reply:
You cannot invent the serialization here, you have to reverse engineer ABAP and implement it the ABAP's way.

Reply:
I understand. I am going to try it.

Reply:
I have found a function group instance (package: `SBAC`, function group: `gui_e2e`) with XML containing:
 ```
 ...
</FUNCTIONS>
<I18N_TPOOL>
    <item>
     <LANGUAGE>1</LANGUAGE>
     <TEXTPOOL>
      <item>
       <ID>id</ID>
       <KEY>key</KEY>
       <ENTRY>entry</ENTRY>
       <LENGTH>40</LENGTH>
      </item>
      <item>
       <ID>id</ID>
       <KEY>key</KEY>
       <ENTRY>entry</ENTRY>
       <LENGTH>80</LENGTH>
      </item>
      ...
     </TEXTPOOL>
    </item>
    <item>
...
```

Notice that `<TEXTPOOL>` is wrapping its items (Structures) in the `<item>` tag. The current XML Serializer cannot create this XML structure (there is no way to differentiate between the case with/without the `<item>` tag; for more details, compare the XML snippet with [FUNCTION](https://github.com/jfilak/sapcli/blob/master/test/unit/fixtures_cli_checkin.py#L121) ).

I thought of two solutions. The first one is the use of immersive InternalTables, where the inner table would signalize the use of the `<item>` tag.  Hence, the code would be:
```
I18N_TPOOL_ITEM = InternalTable.define('I18N_TPOOL_ITEM', STRUCTURE)
I18N_TPOOL = InternalTable.define('I18N_TPOOL', I18N_TPOOL_ITEM)
```
This will affect only the part of the serializer which you have highlighted.

On the other hand, I think that instead of a complicated combination of InternalTables and Structures, a better approach would be to create, for example, `ItemizedTable`, which would be an alias of `InternalTable` to signalize the use of the `<item>` tag. This approach would simplify the XML serialization (the `row_name_getter` would be obsolete --> [used here](https://github.com/jfilak/sapcli/blob/master/sap/platform/abap/__init__.py#L241)). However, there would be no use case for immersive InternalTables (possibly, it can be forbidden). The code would be:
```
I18N_TPOOL = ItemizedTable.define('I18N_TPOOL', STRUCTURE)
```
The downside is that it requires more invasive changes to the whole serializer.

What do you think would be a better approach?

Reply:
abapGit adds the "item" tag randomly when calling zcl_abapgit_xml_output=>add. Hence it is hard to come up with a generic serializer/deserialized and so you need to build the python objects the way you want them to be serialized - that means you have to use a special type allowing the serializer to know when to add "item".